### PR TITLE
fix: Save experiment glide table selection rows in settings [WEB-1366]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

Stores selectAll, selectedExperimentIds, and excludedExperimentIds in persistent settings, so you can leave the new experiment list page and come back with same settings.  These will be neutral values while waiting for `isLoadingSettings`

Previously the source of truth for selection was the GlideTable and data[index] instead of experiment.id, so there are a few changes.
I kept `excludedExperimentIds` as a Set, though it's stored to settings as an array; I am uncertain about whether to keep this a Set and `selectedExperimentIds` being an Array

## Test Plan

**Test on a single page of experiments (latest-main project 107) and a multi-page experiment list (uncategorized)**

Selection
- With `?f_explist_v2=on`, select every other experiment in a list. Refresh or navigate away from the page and return; same experiments should be selected
- The count of selected experiments in the edit button, e.g. "Edit (10)" should reflect the count selected

Select All
- Check the header checkbox on the table - this should select all remaining experiments
- Reload the page; this should keep select-all state, experiments should all be selected, "Edit (10)" should reflect count

Exclusion
- While all are selected, uncheck one row to exclude it.
- Reload the page; this should keep the select-all state, and all should be selected except for the excluded; "Edit (9)" should reflect count

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx [hunk 1]
```diff
@@ -159,17 +159,49 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     }
   }, [settings.filterset, isLoadingSettings]);
 
-  const [selectedExperimentIds, setSelectedExperimentIds] = useState<number[]>([]);
-  const [excludedExperimentIds, setExcludedExperimentIds] = useState<Set<number>>(
-    new Set<number>(),
+  const selectAllIfLoaded = useMemo(
```
Comment (johnkim-det) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:162]: nit: is the addition of `IfLoaded` to these variable names meaningful?

Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:162]: I had named `columnsIfLoaded` in a previous PR to be the default value until settings finish loading. Since it is the value and not Loadable<value>  it may be clearer to drop the IfLoaded?

**Review by johnkim-det — COMMENTED** [review]
On refresh, I'm seeing the state with the expected count (in the Edit button), but the actual rows in the table aren't checked. Trying to "Select all" and "Clear selected" doesn't update the count in the Edit button to 0, either.

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx (hunk 1 above)
Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx]: dropped `IfLoaded` from these variables in a recent commit

**Comment by mapmeld** [comment]
Alright I think the selection issue was due to my testing on project 107 with a single page load of experiments. I've updated the instructions to test on Uncategorized and on a smaller project

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx [hunk 2]
```diff
@@ -159,11 +159,43 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     }
   }, [settings.filterset, isLoadingSettings]);
 
-  const [selectedExperimentIds, setSelectedExperimentIds] = useState<number[]>([]);
```
Comment (ClaireNeveu) [inline webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:162]: 90% of the time we need an `Array<number>`. We only need something else for the initial load and asynchronously saving the settings so this is a bad tradeoff to make. You're slowing down the common case for an uncommon case.

Comment (ClaireNeveu) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:162]: You should set the initial state based on settings otherwise you'll get a blip of incorrect state when navigating to this page from another when when settings are already loaded.

File: webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx [hunk 3]
```diff
@@ -226,15 +220,12 @@ export const GlideTable: React.FC<GlideTableProps> = ({
...
-    setSelectedExperimentIds((prevIds) => {
-      const selectedIds = selectedRowIndices
-        .map((idx) => data?.[idx])
-        .filter((row) => row !== undefined)
-        .filter(Loadable.isLoaded)
-        .map((record) => record.data.experiment.id);
-      if (prevIds === selectedIds) return prevIds;
-      return selectedIds;
-    });
+    const selectedIds = selectedRowIndices
```
Comment (ClaireNeveu) [inline webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx:223]: Why did you change this?

File: webui/react/src/pages/F_ExpList/F_ExperimentList.tsx (hunk 2 above)
Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/F_ExperimentList.tsx:162]: ✅ will init with settings

File: webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx (hunk 3 above)
Comment (mapmeld) [reply webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx:223]: The function `setSelectedExperimentIds` which was passed down from the `useState` in `F_ExperimentList`, got replaced with a new function `handleExperimentSelection` where we also call `updateSettings`. Because we call it with `setSelectedExperimentIds([])` I want this to consistently receive an array (not `(prev) => response` form)

Comment (ClaireNeveu) [reply webui/react/src/pages/F_ExpList/glide-table/GlideTable.tsx:223]: It did consistently receive an array before. Actually, the `if (prevIds === selectedIds) return prevIds;` never fired because it was comparing object ids so I guess this code is equivalent.

**Review by johnkim-det — APPROVED** [review]
LGTM
